src/swell/utilities/slurm.py

```diff
@@ -184,6 +184,12 @@ def prepare_scheduling_dict(
             validate_directives(model_directives)
             scheduling_dict[slurm_task]["directives"][model_component] = model_directives
 
+        # Default execution time limit for everthing is PT1H
+        x = 'PT1H'
+        if slurm_task in experiment_task_directives.keys():
+            x = experiment_task_directives[slurm_task].get('execution_time_limit', x)
+        scheduling_dict[slurm_task]['execution_time_limit'] = x
```

Comment:
Simplify the implementation.

```suggestion
        # Default execution time limit for everthing is PT1H
        if slurm_task in experiment_task_directives.keys():
            x = experiment_task_directives[slurm_task].get('execution_time_limit', x)
        scheduling_dict[slurm_task]['execution_time_limit'] = x
```

Reply:
Hi @ashiklom: 

I kind of missed your message. 
Your modification will miss:
`global set for execution_time_limit` or an `else` statement
because 
L187 -- L191 is inside a do loop and scheduling_dict[slurm_task]["execution_time_limit"] was not previously defined:
for slurm_task in slurm_tasks:
    ...
    ...
    if slurm_task in experiment_task_directives.keys():
            scheduling_dict[slurm_task]["execution_time_limit"] = (
                experiment_task_directives[slurm_task].get("execution_time_limit", "PT1H")
   else:
           scheduling_dict[slurm_task]["execution_time_limit"] = "PT1H"
            without else:  the above execution_time_limit is undefined.
    endif 
end do

These are subtle details, and I donot intend to focus on this and I was trying to preserve the logic in the old code.  Code refactoring to create a clean logic and easy to read manuscript will be great and I am on your side!

Reply:
Ah, I missed that, sorry! We can leave as is.